### PR TITLE
Add registry context to `LootTableLoadEvent`

### LINKED ISSUE #1625: LootTableLoadEvent does not have registry access
**Minecraft Version:** 1.21.1

**NeoForge Version:** 21.1.9

**Description of issue:**

The [`LootTableLoadEvent`](https://github.com/neoforged/NeoForge/blob/d19ca0689a0ef67aa6980645451a0f736ebbb007/src/main/java/net/neoforged/neoforge/event/LootTableLoadEvent.java) appears to be missing some API changes that were made around other loot-table-adjacent things [in 1.20.5](https://gist.github.com/ChampionAsh5357/53b04132e292aa12638d339abfabf955#resourcekey-references); in particular:

* `getName()` returns a `ResourceLocation` and there's no method that returns the New Hotness `ResourceKey<LootTable>` used elsewhere.
    * This is mostly cosmetic, but it's an inconsistency with other API changes,
* There doesn't appear to be any way to access a `HolderLookup.Provider` from this event, which is required to access various registries -- for example, if you want to add an enchantment condition check you need some way to access the enchantment registry.
    * Since this event is fired during deserialization, there is not yet any server OR client, so `CommonHooks.resolveLookup` always fails.
    * The codec (or rather the `RegistryOps`) should have access to the registry lookups, but it needs to be passed into the event.

### PR DESCRIPTION
**Pull request by Matyrobbrt**
Closes #1625
Adds an AT for `RegistryOps.lookupProvider` to be able to extract it from method that only have a registry ops.

### CODE REVIEW

**Review by Shadows-of-Fire — DISMISSED** [review]

File: src/main/java/net/neoforged/neoforge/event/EventHooks.java [hunk 1]
```diff
@@ -688,10 +689,11 @@ public static boolean onProjectileImpact(Projectile projectile, HitResult ray) {
      * which maps to an empty {@link Optional} in {@link LootDataType#deserialize(ResourceLocation, DynamicOps, Object)}
      */
     @Nullable
-    public static LootTable loadLootTable(ResourceLocation name, LootTable table) {
+    @ApiStatus.Internal
+    public static LootTable loadLootTable(ResourceLocation name, LootTable table, HolderLookup.Provider registries) {
         if (table == LootTable.EMPTY) // Empty table has a null name, and shouldn't be modified anyway.
             return null;
-        LootTableLoadEvent event = new LootTableLoadEvent(name, table);
+        LootTableLoadEvent event = new LootTableLoadEvent(registries, name, table);
```
Comment (Shadows-of-Fire) [inline src/main/java/net/neoforged/neoforge/event/EventHooks.java:696]: How did it end up that you added the lookup provider to the end of the args for `loadLootTable` but the front of the args for `LoadLootTableEvent`?

Comment (sciwhiz12) [reply src/main/java/net/neoforged/neoforge/event/EventHooks.java:696]: Why _did_ you put the lookup provide at the end of the hook method but at the start of the event constructor? 🤔

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]

File: src/main/java/net/neoforged/neoforge/event/EventHooks.java (hunk 1 above)
Comment (Matyrobbrt) [reply src/main/java/net/neoforged/neoforge/event/EventHooks.java:696]: What makes you think I willingly put them in different places :P I guess it just happened

File: src/main/java/net/neoforged/neoforge/event/LootTableLoadEvent.java [hunk 2]
```diff
@@ -26,16 +31,36 @@
...
+    /**
+     * {@return a lookup provider that can be used to access registries}
+     */
+    public HolderLookup.Provider getRegistries() {
+        return registries;
+    }
+
     public ResourceLocation getName() {
-        return this.name;
+        return name;
```
Comment (sciwhiz12) [inline src/main/java/net/neoforged/neoforge/event/LootTableLoadEvent.java]: Why _did_ you remove the `this.` qualifier on this field access? 🤔

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]

**Review by sciwhiz12 — APPROVED** [review]

**Pull request merged**